aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -432,38 +429,24 @@ void max_pool2d_with_indices_backward_out_cuda_template(
   checkAllSameGPU("max_pool2d_with_indices_out_cuda",
                   {gradInput_arg, gradOutput_arg, input_arg, indices_arg});
 
-  // #20866, #22032: Guarantee this for the official C++ API?
-  TORCH_CHECK(kernel_size.size() == 1 || kernel_size.size() == 2,
-    "max_pool2d: kernel_size must either be a single int, or a tuple of two ints")
+  TORCH_CHECK(input_.dtype() == gradInput.dtype(),
+    "expected dtype ", input_.dtype(), " for `gradInput` but got dtype ", gradInput.dtype());
```

Comment:
This check also looks like it ought to have been handled by the meta function

Reply:
Nope, if `gradInput` is manually given as input, the process of `set_output` will cause us to check that `input.options()` matches gradInput. So not needed.